### DISCUSSION TITLE
0.23 migration: Snapshot with multiple deployments

### DISCUSSION THREAD

**Discussion by s00500** [pocketbase/pocketbase#5784] [Q&A]
I just followed the migration guide and stumbled into a issue:

To migrate my (jsvm) migrations I deleted them, then created a snapshot, and finally ran history-sync.

This works fine for a single db. But when then pushing migrations and binary to an existing system with a db it will try to reapply the snapshot --- even if I ran history-sync (as that only deletes files that do not exist)

So to fix this I manually added the snapshot to my db migrations table, and this works fine.

I just wanted to highlight that this might be a potential hickup during migrating to 0.23 in the future

(meanwhile thanks for the awesome work here ;-) )

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand what actually is the issue.

When running the new binary and migrations against an existing system, it should:
- apply any new system migrations
- apply any new custom user defined ones (in your case the snapshot)

Or in other words, applying the snapshot to the new system should be expected. Or I'm misunderstanding what do you mean?

**Comment by s00500** [comment]
That is correct. But when migrating a .22 instance to a version that suddenly has migrations deleted and a snapshot as a start it will fail the snapshot migration (at least for me it did fail on a duplicate of _superusers, kind of what I expect...)

**Reply by ganigeorgiev** [reply]
I'll test it before the final release it but I'm still not sure that I understand what actually is the issue.

If the timestamp of your snapshot was > of the system ones _(which is expected to be because the system init migration is with an older timestamp)_ then it should execute as a last migration and it shouldn't fail or change practically anything.

**Comment by s00500** [comment]
This is because the creation of the snapshot would usually also write it into the db. That happens to the specific db where you ran the snapshot, but if that binary + migrations is then pushed to a different deployment you get an issue.

Just a thing that currently one needs to be careful about...
